bindgen/ir/var.rs

```diff
@@ -389,9 +392,94 @@ impl ClangSubItemParser for Var {
     }
 }
 
+fn parse_macro_clang_fallback(
+    ctx: &mut BindgenContext,
+    cursor: &clang::Cursor,
+) -> Option<(Vec<u8>, cexpr::expr::EvalResult)> {
+    let file = ".macro_eval.c".to_string();
+    let contents = format!("int main() {{ {}; }}", cursor.spelling(),);
+
+    let root = if let Some(ftu) = ctx.fallback_translation_unit_mut() {
+        ftu.reparse(&[clang::UnsavedFile::new(&file, &contents)]);
+        ftu.translation_unit().cursor().collect_children()
+    } else {
+        let index = clang::Index::new(false, false);
+
+        let mut c_args = Vec::new();
+        for input_header in ctx.options().input_headers.iter() {
+            let path = Path::new(input_header.as_ref());
+            let header_name = path
+                .file_name()
+                .and_then(|hn| hn.to_str())
+                .map(|s| s.to_owned());
+            let header_path = path
+                .parent()
+                .and_then(|hp| hp.to_str())
+                .map(|s| s.to_owned());
+
+            if let (Some(hp), Some(hn)) = (header_path, header_name) {
+                let header_path = if hp.is_empty() {
+                    Path::new(".")
+                        .canonicalize()
+                        .map(|p| p.display().to_string())
+                        .unwrap_or(".".to_string())
+                } else {
+                    hp.clone()
+                };
+                let pch = format!("{header_path}/{hn}.pch");
+
+                if !Path::new(&pch).exists() {
+                    let header = format!("{header_path}/{hn}");
+                    let mut tu = TranslationUnit::parse(
+                        &index,
+                        &header,
+                        &[
+                            "-x".to_owned().into_boxed_str(),
+                            "c-header".to_owned().into_boxed_str(),
+                        ],
+                        &[],
+                        clang_sys::CXTranslationUnit_ForSerialization,
+                    )?;
+                    tu.save(&pch);
+                }
+
+                c_args.push("-include-pch".to_string().into_boxed_str());
+                c_args.push(pch.into_boxed_str());
+            }
+        }
+
+        OpenOptions::new()
+            .write(true)
+            .create(true)
+            .truncate(true)
+            .open(&file)
+            .ok()?
+            .write_all(contents.as_bytes())
+            .ok()?;
```

Comment:
I wonder, do we really need to write this file to disk? Can we somehow abuse the CXUnsavedFile API to do this in memory in both code paths?

Otherwise, should we delete it?

Also, it's a bit unfortunate that if the user happens to have a valuable thing there (unlikely) we'd throw it away.

Reply:
Please take a look at my most recent addition of `--clang-macro-fallback-build-dir`. That might address your concern.